src/emulator/storage/upload.ts

```diff
@@ -172,13 +176,34 @@ export class UploadService {
    */
   public finalizeResumableUpload(uploadId: string): Upload {
     const upload = this.getResumableUpload(uploadId);
+    if (upload.status === UploadStatus.FINISHED) {
+      throw new UploadPreviouslyFinalizedError();
+    }
     if (upload.status === UploadStatus.CANCELLED) {
       throw new UploadNotActiveError();
     }
     upload.status = UploadStatus.FINISHED;
     return upload;
   }
 
+  /**
+   * Sets previous response code.
+   */
+  public setResponseCode(uploadId: string, code: number): void {
+    const upload = this._uploads.get(uploadId);
+    if (upload) {
+      upload.prevResponseCode = code;
+    }
+  }
+
+  /**
+   * Gets previous response code.
+   * In the case the previous response code doesn't exist (after importing) return 200
```

Comment:
upload ids don't get persisted across import/export

Reply:
yeah thats true, in the case where someone exports and then imports data I was thinking we would just return 200 when finalize is called twice. 

It didn't seem to make sense to persist this previous response code on export since that would mean adding it to the metadata which seemed strange and different from prod metadata.

In any case returning 200 here would keep the emulator from crashing and as far as I know theres no equivalent flow to this import/export on prod so this wouldn't be introducing any divergence.